Autogens to enable kde6

linuxexplorer opened the pull request:
Bugs: macaroni-os/mark-issues#537
All new ebuilds have been tested locally and merge.

lichess4545stats committed:
Autogens to enable kde6

Bugs: macaroni-os/mark-issues#537
All new ebuilds have been tested locally and merge.

geaaru commented on core-hw-kit/autogen.kit.d/base.yml:
```diff
@@ -282,3 +282,114 @@ upower:
+            - meson
+            - python-single-r1
+            - systemd
+          iuse: bash-completion gtk-doc man selinux systemd test zsh-completion
```
`test` could be removed together with the dependencies on BDEPEND and `python_check_deps` and `src_test`

geaaru commented on core-hw-kit/autogen.kit.d/base.yml:
```diff
@@ -282,3 +282,114 @@ upower:
+            src_install() {
+              meson_src_install
+              python_fix_shebang "${ED}"/usr/bin/powerprofilesctl
+              newinitd "${FILESDIR}/power-profiles-daemon.initd" power-profiles-daemon
```
I'm not sure if it's really needed a initd daemon for this.
Alpine just patches the .dbus.service.in. See https://gitlab.alpinelinux.org/alpine/aports/-/blob/master/community/power-profiles-daemon/0001-dbus-no-systemd.patch

We can try when the ebuild is created.

geaaru commented on kde-kit/autogen.kit.d/base.yml:
```diff
@@ -0,0 +1,488 @@
+                        $(qca_plugin_use softstore)
+                        $(qca_plugin_use ssl ossl)
+                        -DBUILD_TESTS=$(usex test)
+                )
```
`test` to remove and probably indentation issues?

geaaru commented on kde-kit/autogen.kit.d/base.yml:
```diff
@@ -0,0 +1,488 @@
+    - qcoro:
+        vars:
+            license: MIT
+            slot: '0'
```
This is the default. You can remove it

linuxexplorer replied:
Are you referring just to `slot: '0'`?

geaaru replied:
Yeah, it's the default. You can drop the line. ;)

geaaru commented on kde-kit/autogen.kit.d/base.yml:
```diff
@@ -0,0 +1,488 @@
+          desc: VLC backend for the Phonon multimedia library
+          body: |
+            MY_PN="phonon-backend-vlc"
+            S="${WORKDIR}/${MY_PN}-${PV}"
```
You can replace `MY_PN` and  `S`.
1) `S` could be defined as `s` vars eventually.
2) probably you can replace boh with this:
```yaml
vars:
   s: '${WORKDIR}/phonon-backend-vlc-{{ .Values.version }}'
```

geaaru commented on kde-kit/autogen.kit.d/base.yml:
```diff
@@ -0,0 +1,488 @@
+              cmake_src_configure
+            }
+
+pulseaudio_qt:
```
With last releases of mark-devkit you can use Helm template over transform and matches. So probably you can use a single definition for both `pulseaudio_qt` and `polkit_qt` considering that using the same url.

So, something like this:
```yaml
# Example
mydef:
    generator: builtin-dirlisting
    defaults:
        dir:
           url: 'https://download.kde.org/stable/{{ .Values.pkgname | default .Values.pn }}/'
           matcher: '{{ .Values.pkgname | default .Values.pn }}-[0-9]+.tar.xz$'
         transform:
            - kind: string
              match: '{{ .Values.pkgname | default .Values.pn }}-'
              replace: ''
            - kind: string
              match: '.tar.xz'
              replace: ''
     packages:
        - pulseaudio-qt:
             ...
        - polkit-qt:
            ...
```

geaaru commented on net-kit/autogen.kit.d/base.yml:
```diff
@@ -2627,4 +2627,159 @@ gitlab_freedesktop_dirlisting:
+      inherit:
+        - libtool
+      iuse: dbus exif jpeg +poppler +postscript png test tiff
+      restrict: '!test? ( test )'
```
Also here `test` could be removed and set to `false` statically in the configure

geaaru commented on net-kit/autogen.kit.d/base.yml:
```diff
@@ -2627,4 +2627,159 @@ gitlab_freedesktop_dirlisting:
+      repo: libcupsfilters
+    assets:
+      - name: '{{ .Values.pn }}-{{ .Values.version }}.tar.xz'
+        url: 'https://github.com/OpenPrinting/{{ .Values.pn }}/releases/download/{{ .Values.version }}/{{ .Values.pn }}-{{ .Values.version }}.tar.xz'
```
Cosmetic: When there assets and `builtin-github` you can also just use `matcher` of the tarball instead of the full path. See for example: https://github.com/macaroni-os/kit-fixups/blob/mark-unstable/core-server-kit/autogen.kit.d/telemetry.yml#L17.

It's fine this way too. Just sharing an alternative way for github packages.

geaaru commented on kde-kit/autogen.kit.d/qca/qca-disable-pgp-test.patch:
```diff
@@ -0,0 +1,13 @@
+ add_subdirectory(macunittest)
+ add_subdirectory(metatype)
+-add_subdirectory(pgpunittest)
++# add_subdirectory(pgpunittest)
```
I think that this Gentoo-style patch could be replaced with a simple sed in `src_prepare`.
Something, like this:
```
sed -e '/^add_subdirectory\(pgpunittest/d' unittest/CMakeLists.txt -i || die
```

lichess4545stats committed:
Autogens to enable kde6

Bugs: macaroni-os/mark-issues#537

linuxexplorer committed:
Autogens to enable kde6

Added cups-filters and cups-browsed autogens, for completeness (note:
braille-printer-app does not appear to have releases so not included).

geaaru commented on dev-kit/autogen.kit.d/base.yml:
```diff
@@ -1850,3 +1850,139 @@ appstream:
+        - cmake
+        - flag-o-matic
+        - prefix
+      iuse: cpu_flags_x86_sse2 curl doc examples iconv icu static-libs test threads
```
`test` could be removed

geaaru commented:
Moving ahead this PR that fixes also:
* https://github.com/macaroni-os/mark-issues/issues/138
* https://github.com/macaroni-os/mark-issues/issues/153

geaaru commented:
Thank you very much
